src/Uno.UWP/Data/Pdf/PdfDocument.Android.cs

```diff
@@ -0,0 +1,112 @@
+	{
+		return LoadFromFileAsync(file, null);
+	}
+
+	public static IAsyncOperation<PdfDocument> LoadFromFileAsync(Storage.IStorageFile file, string password)
+	{
+		if (!string.IsNullOrWhiteSpace(password))
+		{
+			// password protected PDF files are not supported by PdfRenderer in Android
+			throw new NotImplementedException("The member IAsyncOperation<PdfDocument> PdfDocument.LoadFromFileAsync(IStorageFile file, string password) is not implemented in Uno.");
+		}
+
+		if (file is null)
+		{
+			throw new ArgumentNullException(nameof(file));
+		}
+
+		var localpath = file.Path;
+		var fileDescriptor = localpath.StartsWith('/')
+			? ParcelFileDescriptor.Open(new Java.IO.File(localpath), ParcelFileMode.ReadOnly)
+			: Android.App.Application.Context.ContentResolver.OpenFileDescriptor(Android.Net.Uri.Parse(localpath), "r");
+		var pdfRenderer = new PdfRenderer(fileDescriptor);
+		var pdfDocument = new PdfDocument(pdfRenderer);
+
+		return Task.FromResult(pdfDocument).AsAsyncOperation();
+	}
+
+	public static IAsyncOperation<PdfDocument> LoadFromStreamAsync(Storage.Streams.IRandomAccessStream inputStream)
+	{
+		return LoadFromStreamAsync(inputStream, default);
+	}
+
+	public static IAsyncOperation<PdfDocument> LoadFromStreamAsync(Storage.Streams.IRandomAccessStream inputStream, string password)
+	{
+		if (!string.IsNullOrWhiteSpace(password))
+		{
+			// password protected PDF files are not supported by PdfRenderer in Android
+			throw new NotImplementedException("The member IAsyncOperation<PdfDocument> PdfDocument.LoadFromStreamAsync(IRandomAccessStream inputStream, string password) is not implemented in Uno.");
+		}
+		return AsyncOperation.FromTask(async ct =>
+		{
+			var parcel = await GetParcelFileDescriptorFromStreamAsync(inputStream.AsStream(), ct);
+			var pdfRenderer = new PdfRenderer(parcel);
+			var pdfDocument = new PdfDocument(pdfRenderer);
+			return pdfDocument;
+		});
+	}
+
+	public void Dispose()
+	{
+		_pdfRenderer.Dispose();
+	}
+
+	private static Task<ParcelFileDescriptor> GetParcelFileDescriptorFromStreamAsync(Stream input, CancellationToken token) =>
+		Task.Run(() =>
+			{
+				/* PdfRenderer https://developer.android.com/reference/android/graphics/pdf/PdfRenderer#PdfRenderer(android.os.ParcelFileDescriptor)
+				 * PdfRenderer needs a seek-able ParcelFileDescriptor,
+				 * to get it from a stream (without using hacks),
+				 * you need to copy the stream to a temporary file.
+				*/
+				var fileName = Path.GetTempFileName();
+				var file = new Java.IO.File(fileName);
+				file.DeleteOnExit();
+				var outputStream = new Java.IO.FileOutputStream(file);
+
+				byte[] buf = new byte[4096];
+				var len = 0;
+				while (!token.IsCancellationRequested && (len = input.Read(buf, 0, buf.Length)) > 0)
+				{
+					outputStream.Write(buf, 0, len);
+				}
+				outputStream.Flush();
+				outputStream.Close();
+				return ParcelFileDescriptor.Open(file, ParcelFileMode.ReadOnly);
+			}, token);
```

Comment:
Why this returns a Task? I think this can just return `ParcelFileDescriptor`

Reply:
Because with very large pdfs, copying the stream to the temporary file could take a long time.